PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Unshipped.txt

```diff
@@ -1,3 +1,5 @@
+Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration.ApplicationInsightsEndpointBaseAddress.set -> void
```

Comment:
We planned to remove EndpointAddress from `ITelemetryChannel` interface. Not every channel has an endpoint - some are ETW or has more parameters than just URL. So I don't think it's a good idea to add a base address here.

`app-id` resolution is needed by dependency telemetry module. Can this property be defined there instead?

Reply:
No. CorrelationIdLookupHelper is used by more than just dependency telemetry module.
The intent is to simplify our user configuration story by moving this property to the root of the config to allow all of our Microsoft services to inherit from this single property.

Reply:
The problem with endpoint approach is that this approach doesn't work for non-http-based extraction of app-id. What if we want app-id to be set from config? Or read from registry key that set by Agent?

I don't like the idea of implementation detail for our specific service creep out into the base library.

I agree though it is a good idea to have some identity resolution mechanism to be defined in here. Is there an issue with the design discussion?

Reply:
We currently don't have an interface for app-id resolution.
I feel like that would be a separate discussion from this change (which I'm open to having :) )

I agree with the hesitation towards implementation specific details in the BaseSdk.
But I feel like this approach is the best compromise to the alternatives. 
Lets discuss this in more detail.